demonstrations/tutorial_rosalin.py

```diff
@@ -0,0 +1,598 @@
+r"""
+Frugal shot optimization with Rosalin
+=====================================
+
+.. meta::
+    :property="og:description": In this demonstration, the Rosalin optimizer
+    	is used to minimize a Hamiltonian via an frugal shot optimization strategy.
+    :property="og:image": https://pennylane.ai/qml/_images/single_shot.png
+
+In this tutorial we investigate and implement the Roaslin (Random Operator Sampling for
+Adaptive Learning with Individual Number of shots) from
+Arrasmith et al. [#arrasmith2020]_. In this paper, a strategy
+is introduced for reducing the number of shots required when optimizing variational quantum
+algorithms, by randomly sampling operators from the cost Hamiltonian.
+
+Background
+----------
+
+While a large number of results in variational quantum algorithms focus on the
+choice of circuit ansatz, cost function, gradient computation, or initialization strategy,
+the optimization strategy --- an important component affecting both convergence time and
+quantum resource dependence --- is not as frequently considered. Instead, common
+'out-of-the-box' classical optimization techniques, such as gradient-free
+(COBLYA, Nelder-Mead), gradient-descent, and Hessian-free (L-BFGS) tend to be used.
+
+However, for variational algorithms such as :doc:`VQE </demos/tutorial_vqe>`, which involve evaluating
+a large number of non-commuting operators in the cost function, decreasing the number of
+quantum evaluations required for convergence, while still minimizing shot noise, can

```

Comment:
Think we want to avoid "shot noise"
```suggestion
quantum evaluations required for convergence, while still minimizing statistical noise, can

```